packages/devtools_app/lib/src/extension_points/extensions_external.dart

```diff
@@ -49,44 +49,43 @@ class ExternalDevToolsExtensionPoints implements DevToolsExtensionPoints {
 
 const _newDevToolsIssueUriDisplay = 'github.com/flutter/devtools/issues/new';
 
-Uri _newDevToolsGitHubIssueUri({String? issueDetails}) {
+Uri _newDevToolsGitHubIssueUriLengthSafe({String? issueDetails}) {
   const _maxGitHubUriLength = 8190;
+  final context = issueLinkDetails();
 
+  final fullUri =
+      _newDevToolsGitHubIssueUri(issueDetails: issueDetails, context: context);
+
+  final lengthToCut = fullUri.toString().length - _maxGitHubUriLength;
+  if (lengthToCut <= 0) return fullUri;
+
+  if (issueDetails == null)
+    throw StateError(
+      'Issue details cannot be null, because length limit is reached.',
+    );
```

Comment:
this error is a bit confusing 🤔  instead of throwing an error, lets add the check for `issueDetails == null` to the if check on line 60.

`if (issueDetails == null || lengthToCut <=0) return fullUri;`